compiler/rustc_mir_transform/src/dataflow_const_prop.rs

```diff
@@ -266,15 +264,16 @@ impl<'tcx> ValueAnalysis<'tcx> for ConstAnalysis<'_, 'tcx> {
                     FlatSet::Top => FlatSet::Top,
                 }
             }
-            Rvalue::BinaryOp(op, box (left, right)) => {
+            Rvalue::BinaryOp(op, box (left, right)) if !op.is_overflowing() => {
                 // Overflows must be ignored here.
+                // The overflowing operators are handled in `handle_assign`.
```

Comment:
I don't understand *why* the overflowing operators can't be handled here, but it matches the previous structure where `Rvalue::CheckedBinaryOp` was handled in `handle_assign`, not in `handle_rvalue`, for some reason.

Reply:
As checked ops return scalar pairs, the routine needs to project the destination as 2 scalars, which `handle_rvalue` is not designed to do.

Reply:
Would be good to add a comment in the code explaining that.